stac_pydantic/api/search.py

```diff
@@ -38,96 +31,65 @@
 SearchDatetime = TypeAdapter(Optional[UtcDatetime])
 
 
-class Search(BaseModel):
-    """
-    The base class for STAC API searches.
+def validate_bbox(v: Optional[BBox]) -> Optional[BBox]:
+    """Validate BBOX value."""
+    if v:
+        # Validate order
+        if len(v) == 4:
+            xmin, ymin, xmax, ymax = cast(Tuple[int, int, int, int], v)
 
-    https://github.com/radiantearth/stac-api-spec/blob/v1.0.0/item-search/README.md#query-parameter-table
-    """
+        elif len(v) == 6:
+            xmin, ymin, min_elev, xmax, ymax, max_elev = cast(
+                Tuple[int, int, int, int, int, int], v
+            )
+            if max_elev < min_elev:
+                raise ValueError(
+                    "Maximum elevation must greater than minimum elevation"
+                )
+        else:
+            raise ValueError("Bounding box must have 4 or 6 coordinates")
 
-    collections: Optional[List[str]] = None
-    ids: Optional[List[str]] = None
-    bbox: Optional[BBox] = None
-    intersects: Optional[Intersection] = None
-    datetime: Optional[str] = None
-    limit: Optional[int] = 10
+        # Validate against WGS84
+        if xmin < -180 or ymin < -90 or xmax > 180 or ymax > 90:
+            raise ValueError("Bounding box must be within (-180, -90, 180, 90)")
 
-    # Private properties to store the parsed datetime values. Not part of the model schema.
-    _start_date: Optional[dt] = PrivateAttr(default=None)
-    _end_date: Optional[dt] = PrivateAttr(default=None)
+        if ymax < ymin:
+            raise ValueError("Maximum longitude must be greater than minimum longitude")
 
-    # Properties to return the private values
-    @property
-    def start_date(self) -> Optional[dt]:
-        return self._start_date
+    return v
 
-    @property
-    def end_date(self) -> Optional[dt]:
-        return self._end_date
 
-    # Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
-    @model_validator(mode="before")
-    def validate_spatial(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        if values.get("intersects") and values.get("bbox") is not None:
-            raise ValueError("intersects and bbox parameters are mutually exclusive")
-        return values
+def str_to_datetimes(value: str) -> List[Optional[dt]]:
```

Comment:
Might be a breaking change so might not be appropriate here, but this should be a two-tuple, yeah? 


```suggestion
def str_to_datetimes(value: str) -> tuple[Optional[dt], Optional [dt]]:
```

Reply:
`cast` will return a `List` of datetime|None. So as it is right now is fine. 

It's not necessary a two element tuple. It could be a list of one element (if datetime=single-datetime)